pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -160,15 +172,48 @@ def _run_early_stopping_check(self, trainer):
         # when in dev debugging
         trainer.dev_debugger.track_early_stopping_history(self, current)
 
-        if self.monitor_op(current - self.min_delta, self.best_score):
+        should_stop, reason = self._evalute_stopping_criteria(current)
+
+        # stop every ddp process if any world process decides to stop
+        should_stop = trainer.training_type_plugin.reduce_boolean_decision(should_stop)
+        trainer.should_stop = trainer.should_stop or should_stop
+        if should_stop:
+            self.stopped_epoch = trainer.current_epoch
+        if reason:
+            log.info(f"[{trainer.global_rank}] {reason}")
+
+    def _evalute_stopping_criteria(self, current: torch.Tensor) -> Tuple[bool, str]:
+        should_stop = False
+        reason = None
+        if self.check_finite and not torch.isfinite(current):
+            should_stop = True
+            reason = (
+                f"Monitored metric {self.monitor} = {current} is not finite."
+                f" Previous best value was {self.best_score:.3f}. Signaling Trainer to stop."
+            )
+        elif self.stopping_threshold is not None and self.monitor_op(current, self.stopping_threshold):
+            should_stop = True
+            reason = (
+                f"Below tolerance {self.monitor} = {current} <= {self.stopping_threshold}."
+                " Signaling Trainer to stop."
+            )
+        elif self.divergence_threshold is not None and self.monitor_op(-current, -self.divergence_threshold):
+            should_stop = True
+            reason = (
+                f"Divergence: {self.monitor} = {current} > {self.divergence_threshold}."
```

Comment:
Minor point:  whether it is
```
f"Divergence: {self.monitor} = {current} > {self.divergence_threshold}."
```
or
```
f"Divergence: {self.monitor} = {current} > {self.divergence_threshold}."
```
should depend on the `monitor_op`, right?  Maybe have a `f"Divergence: {self.monitor} = {current} {op_to_string(self.monitor_op)} {self.divergence_threshold}."`
or something like that, where you fill `op_to_string` with whatever you need to turn it into a `>`, etc.?

Similarly, I think you could do the same with the  "successful" convergence below/above the target.
```python
 f"Below tolerance {self.monitor} = {current} {op_to_string(self.monitor_op)} {self.stopping_threshold}  ."
```
where you would have to ensure the order is correct as it is going form the other direction.

As I said though, minor.